### PR TITLE
Added empty constructors to TypedArrays from es2017

### LINKED ISSUE #19680: Empty constructors for TypedArrays are not allowed
Typescript currently does not support empty constructors of TypedArrays, as in [mdn UInt8Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/UInt8Array), thx to [sindresorhus](https://github.com/sindresorhus/is/pull/28#discussion_r148431869) for pointing that out.

**TypeScript Version:**  2.7.0-dev.201xxxxx

**Code**

```ts
const x = new Uint8Array()
```

**Expected behavior:**
works
**Actual behavior:**
Error: Expected 1-3 arguments, but got 0.

### PR DESCRIPTION
**Pull request by timocov**
Fixes #19680

### CODE REVIEW

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -141,6 +141,7 @@ namespace ts {
                     "es2017.sharedmemory": "lib.es2017.sharedmemory.d.ts",
                     "es2017.string": "lib.es2017.string.d.ts",
                     "es2017.intl": "lib.es2017.intl.d.ts",
+                    "es2017.typedarrays": "lib.es2017.typedarrays.d.ts",
```
Comment (mhegazy) [inline src/compiler/commandLineParser.ts]: mind updating it in https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Compiler%20Options.md as well.

Comment (timocov) [reply src/compiler/commandLineParser.ts]: https://github.com/Microsoft/TypeScript-Handbook/pull/680

**Comment by mhegazy** [comment]
merged manually. thanks!

File: tests/cases/conformance/es2017/useTypedArrays1.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// @target: es5
```
Comment (mhegazy) [inline tests/cases/conformance/es2017/useTypedArrays1.ts]: for future reference, no need for the tests for changes to lib like these.

Comment (timocov) [reply tests/cases/conformance/es2017/useTypedArrays1.ts]: I took it from neighbor files in `tests/cases/conformance/es2017` but I will keep it on mind

**Pull request closed without merging**